src/onegov/town6/forms/settings.py

```diff
@@ -143,5 +172,71 @@ def populate_chat_staff(self):
             (v, k) for v, k in staff_members
         ]
 
+    def validate(self):
+        result = super().validate()
+        if self.specific_opening_hours.data:
+            if not self.json_to_time(self.opening_hours_chat.data):
+                self.opening_hours_chat.errors.append(
+                    _('Please add a day and times to each opening hour '
+                      'entry or deactivate specific opening hours.')
+                )
+                result = False
+            for day, start, end in self.json_to_time(
+                self.opening_hours_chat.data
+            ):
+                if not (day and start and end):
+                    self.opening_hours_chat.errors.append(
+                        _('Please add a day and times to each opening hour '
+                          'entry or deactivate specific opening hours.')
+                    )
+                    result = False
+                if start > end:
+                    self.opening_hours_chat.errors.append(
+                        _("Start time cannot be later than end time.")
+                    )
+                    result = False
+        return result
+
+    def json_to_time(self, text=None):
+        result = []
+
+        for value in json.loads(text or '{}').get('values', []):
+            result.append([value.get('day', ''), value.get('start', ''),
+                          value.get('end', '')])
+
+        return result
+
+    def time_to_json(self, opening_hours=None):
+        opening_hours = opening_hours or []
+
+        return json.dumps({
+            'labels': {
+                'day': self.request.translate(_("day")),
+                'start': self.request.translate(_("Start")),
+                'end': self.request.translate(_("End")),
+                'add': self.request.translate(_("Add")),
+                'remove': self.request.translate(_("Remove")),
+            },
+            'values': [
+                {
+                    'day': o[0],
+                    'start': o[1],
+                    'end': o[2],
+                    'error': self.time_errors.get(ix, "")
+                } for ix, o in enumerate(opening_hours)
+            ],
+            'days': {
+                0: self.request.translate(_("Mo")),
+                1: self.request.translate(_("Tu")),
+                2: self.request.translate(_("We")),
+                3: self.request.translate(_("Th")),
+                4: self.request.translate(_("Fr")),
+                5: self.request.translate(_("Sa")),
+                6: self.request.translate(_("Su"))
+            }
+        })
+
     def on_request(self):
         self.populate_chat_staff()
+        for include in self.on_request_include:
+            self.request.include(include)
```

Comment:
We put this here in an experiment, I suppose it's not necessary now that we used the linking from org approach.
```suggestion
```

Reply:
Good point, I'll try it without.